assembly/std/storage.ts

```diff
@@ -6,11 +7,23 @@ import {Address} from './address';
  *
  * Note: Existing entries are overwritten and missing ones are created.
  *
- * @param {StaticArray<u8>} key
- * @param {StaticArray<u8>} value
+ * @param {T} key
+ * @param {T} value
  */
-export function set(key: StaticArray<u8>, value: StaticArray<u8>): void {
-  env.set(key, value);
+export function set<T>(key: T, value: T): void {
+  let keyByte =
+    key instanceof String
+      ? toBytes(key.toString())
+      : key instanceof StaticArray<u8>
+      ? (key as StaticArray<u8>)
+      : new StaticArray<u8>(0);
```

Comment:
This code is too complex and dangerous: if you call set with something else than a string or a byte array, it will compile but failed at the execution because you will send an empty array for the key and the value.

The FFI concept is complex enough and I don't want to build magic stuff that doesn't work 100% of the time over it.

Reply:
Yeah I agree with what you say, didn't notice that.

And what about if we abort when a key OR value is not a string or a byte array ? 

The other solution would be to build specific functions : 
- setOfString 
- getString 
....
But we'll multiply the amount differents functions exposed by the storage interface

Reply:
On this specific subject, @Thykof had a very good answer here : https://discord.com/channels/721472913886281818/1047897635018780682